Code/src/main/java/nl/utwente/group10/ui/BackendUtils.java

```diff
@@ -31,4 +33,18 @@ public static Type dive(ConstT function, int steps) {
         }
         return function;
     }
+    
+    /**
+     * @return Whether or not t1 and t2 unify with each other.
```

Comment:
You didn't receive a concise answer because there is no concise answer.

Here's the thing. Haskell uses global type inference to infer the types of its expressions. The Haskell compiler takes the functions you wrote and the functions in its library, and for every function call, every constant, every *anything* tries to come up with a sensible type for it, one that fits with all the rest of the program. 

To put it another way, the only way to check a program to do type inference on it. This will also, as a side effect, tell you whether or not a program is well-typed: If type inference works, you have a well-typed program. As a bonus, you also have a reasonable idea of what the types are going to be. If type inference fails, you don't have a well-typed program.

This sounds like an NP-complete problem, but joy of joys, there are algorithms available to do this kind of inference. Type inference is quite a complicated subject, but we were able to make it work because we picked one that's well-documented (Hindley-Damas-Milner) and implemented a *subset* of that algorithm for the *subset* of Haskell we target, which is relatively simple. 

The end result is that you have a machine available to you that does just that. It takes a whole program and does type inference on it, leaving you with a typed, checked program. It will even let you know if there's a problem.

JJ and I have been working quite long and hard on thinking it through, getting it to work, and testing it under different circumstances. It pains us that you apparently don't want to use or understand it.